### ISSUE TITLE
ConcurrentWrites=true crashes on UWP without exception

### ISSUE THREAD

**Issue by mlischka** [NLog/NLog#2742]
**Type**: Bug
**NLog version**: 4.5.4.7700 (NLog for NetStandard 2.0)
**Platform**: UWP 6.1.4 on Windows 1803 (same behavior with managed and native)
**Current NLog config**: `FileTarget` with `KeepFileOpen=true`

- *What is the current result?*
While using a `FileTarget` with `KeepFileOpen=true` the app crashs immediately without an exception/stacktrace when the first log entry should be written. 

- *Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?*
Yes, the logger just shuts down after trying to create an appender.
```
...
2018-05-25 13:29:53.4018 Debug FileTarget(Name=): Process file 'C:\Users\Developer\AppData\Local\Packages\46eb4355-0396-4686-a706-046c3ba94973_4ca7txfpdd20p\LocalState\Log\log.txt' on startup
2018-05-25 13:29:53.4149 Debug FileTarget(Name=): Do archive file: 'C:\Users\Developer\AppData\Local\Packages\46eb4355-0396-4686-a706-046c3ba94973_4ca7txfpdd20p\LocalState\Log\log.txt'
2018-05-25 13:29:53.4451 Debug Creating file appender: C:\Users\Developer\AppData\Local\Packages\46eb4355-0396-4686-a706-046c3ba94973_4ca7txfpdd20p\LocalState\Log\log.txt
2018-05-25 13:29:53.5861 Info Shutting down logging...
2018-05-25 13:29:53.5957 Trace Flushing all targets...
...
```
- *Are there any workarounds?*
When setting `KeepFileOpen=false`, all works fine. Is this a bug or limitation when using NLog on UWP?

I traced this down the line and think the cause is that `MutexMultiProcessFileAppender` and `WindowsMultiProcessFileAppender` e.g. `BaseMutexFileAppender` creates a **global** mutex, but those are [not supported on UWP](https://msdn.microsoft.com/en-us/library/aa382954(v=vs.85).aspx) (*Note: The global namespace is not available for Windows Store apps.*) - [are they](https://github.com/dotnet/corefx/issues/20274#issuecomment-303874457)?

When I just execute `new Mutex(false, "Gloabl\anything")` on an UWP app, I get the same behavior, the app crashs.

Is there any way to use `KeepFileOpen=true` on UWP?

**Comment by snakefoot** [comment]
UWP is special just like Android and iOS. NLog tries to cater for this by removing all traps in NetStandard13.

But when building UWP-application that "supports" NetStandard2.0 then protection is gone. NLog expects NetStandard20 applications (Like AspNetCore) to have full support for all the good stuff.

The workaround for you is to configure both `KeepFileOpen=true` and `concurrentWrites=false`

**Comment by snakefoot** [comment]
@304NotModified Rename is issue to "`concurrentWrites=true` crashes on UWP without exception" ?

**Comment by 304NotModified** [comment]
:+1:

**Comment by snakefoot** [comment]
See also: #3048 about changing default value to `concurrentWrites="false"` for better platform support. (UWP / Xamarin apps are optimized not know about global mutex)

**Comment by snakefoot** [comment]
Resolved by #3048
